### PR TITLE
Fix generation of translation parent uid in index

### LINKED ISSUE #504: Translated events not displayed on translated page (with language setting = "strict")
I don't know if I am missing something. I tested this on 2 systems, one with the migrated content. On the other with events created from scratch.

## System 

* calendarize: 8.1.0 (also reproducible with "master")
* TYPO3: 10.4.13

## Reproduce

1. create storage page with some events
2. localize the events (translate)
3. create plugin List+Detail on one page, leave most configuration as default
4. localize the page, copy the plugin (or set language in plugin to -1: has same result).

Load translated page in Frontent. No events are displayed. Events are displayed for default language though. 

## Problem

On the translated page, no events are displayed. Only if I create an event with language = -1, the event is displayed.

### PR DESCRIPTION
**Pull request by okmiim**
- Fix #504 - Fix generation of translation parent uid in index
  - The Reindex scheduler should be run afterwards
- Remove unused fields from index 
  - l10n_diffsource
  - editlock
- Minor fixes and improvements

### CODE REVIEW

**Comment by lochmueller** [comment]
Hey @okmiim 
Thanks again for your work 👍 
Merged...
Regards,
Tim

File: Configuration/TCA/tx_calendarize_domain_model_index.php [hunk 1]
```diff
@@ -8,6 +8,15 @@
 
 $base = ModelUtility::getTcaInformation(Index::class);
 
+// Removed unused fields, because the records are not accessible by the user
+unset(
```
Comment (sypets) [inline Configuration/TCA/tx_calendarize_domain_model_index.php:12]: I am not sure of the unset here - even if fields are not accessible by the user they can still have TCA definitions. In some cases this is advisable to set default values. If the editlock is not accessible, shouldn't it get removed entirely?

**Review by sypets — COMMENTED** [review]
Hi, tested with this patch on our testsystem and it is now working with "strict" (after flushing caches and reindexing). Thanks you!

Will it be generally necessary to redindex or only on systems with non-default language. Should you add an upgrade wizard? Could be purely informative (like the argon2i upgrade wizard) only to inform about redindexing, maybe make a quick check if this should be necessary.

File: Configuration/TCA/tx_calendarize_domain_model_index.php (hunk 1 above)
Comment (lochmueller) [reply Configuration/TCA/tx_calendarize_domain_model_index.php:12]: The editlock field is added by EXT:autoloader. In EXT:autoloader it is part of the "enableFields" and we cannot disable a single field out of this set: https://github.com/lochmueller/autoloader/blob/master/Classes/DataSet/EnableFields.php

Comment (okmiim) [reply Configuration/TCA/tx_calendarize_domain_model_index.php:12]: I missed that point. The column definition of editlock can also be removed.
```php
unset(
    $base['columns']['editlock'],
);
```
---
The reindex in only required, if multiple languages are used.

**Pull request merged**